ACE/ace/CDR_Base.cpp

```diff
@@ -948,8 +951,8 @@ void ACE_CDR::Fixed::normalize (UShort min_scale)
 
 ACE_CDR::Fixed ACE_CDR::Fixed::from_string (const char *str)
 {
-  const bool negative = str && *str == '-';
-  if (negative || (str && *str == '+'))
+  const bool negative = *str == '-';
```

Comment:
It seems like this removes a valid check for zero-valued str pointer, which is probably a bad idea. In fact, CID 1297401 complains that a zero-valued str is passed to strspn, below. It seems better to check for zero str before doing any of these other checks and bail out right away.